File: 프로그래머스/0/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.js
function solution(board, k) {
    var answer = 0;
    
    for (let i = 0; i < board.length; i++) {
        for (let j = 0; j < board[i].length; j++) {
            if (i + j <= k) {
            // 만약 i + j <= k일때
                answer += board[i][j];
            }   // answer에 조건을 만족하는 모든 board[i][j]의 값을 누적합산한다
        }
    }
    return answer;
}

File: 프로그래머스/unrated/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.java
class Solution {
    public int solution(int[][] board, int k) {
        int answer = 0;
        
        for(int i = 0; i < board.length; i++) {
			for(int j = 0; j < board[i].length; j++) {
				if(i + j <= k) { 
                // 만약 i + j <= k일때
					answer += board[i][j];
				}  // answer에 조건을 만족하는 모든 board[i][j]의 값을 누적합산한다
			}
		}
        return answer;
    }
}